bin/trollstalker.py

```diff
@@ -141,17 +141,66 @@ def process(self, event):
         if not event.dir:
             LOGGER.debug("processing %s", event.pathname)
             # parse information and create self.info OrderedDict{}
-            self.parse_file_info(event)
-            if len(self.info) > 0:
-                # Check if this file has been recently dealt with
-                if event.pathname not in self._deque:
-                    self._deque.append(event.pathname)
-                    message = self.create_message()
-                    LOGGER.info("Publishing message %s", str(message))
-                    self.pub.send(str(message))
+
+            # check if event is triggered on a ref file
+            #   if filesize<1kb it is a ref file
+            file_to_check = event.pathname
+            try:
+                file_check_size = os.path.getsize(file_to_check)
+                ref_file_parse = self.file_parser.parse(file_to_check)
+                if(file_check_size<=1000):
+                    # if event is on a ref file, read inside the referenced directories
+                    """REF file internal format:
+                       [REF]
+                       SourcePath = /path/to/dataset
+                       FileName = ref_file_filename
+                    """
+                    LOGGER.info("Found ref file: {}".format(file_to_check))
+                    try:
+                        reffile = RawConfigParser()
+                        reffile.read(file_to_check)
+                    except:
+                        LOGGER.error("Wrong ref file format: " + str(file_to_check))
+                        pass
```

Comment:
So even if the file format is wrong, the function continues ?!?

Reply:
modified: now thread execution continues, but not the function execution.